.github/workflows/swift-toolchain.yml

```diff
@@ -700,6 +698,123 @@ jobs:
             ${{ steps.export-binary-paths.outputs.swift_serialize_diagnostics }}
             ${{ steps.export-binary-paths.outputs.swift_compatibility_symbols }}
 
+  early_swift_driver:
+    # TODO: Build this on Windows too.
+    if: inputs.build_os == 'Darwin'
+    needs: [sqlite]
+    runs-on: ${{ inputs.default_build_runner }}
+
+    strategy:
+      fail-fast: false
+      matrix: ${{ fromJSON(inputs.host_matrix) }}
+
+    name: ${{ matrix.os }} ${{ matrix.arch }} Early Swift Driver
+
+    steps:
+      - uses: actions/download-artifact@v4
+        with:
+          name: sqlite-${{ matrix.os }}-${{ matrix.arch }}-${{ inputs.swift_toolchain_sqlite_version }}
+          path: ${{ github.workspace }}/BuildRoot/Library/sqlite-${{ inputs.swift_toolchain_sqlite_version }}/usr
+
+      - uses: actions/checkout@v4
+        with:
+          repository: swiftlang/swift-llbuild
+          ref: ${{ inputs.swift_llbuild_revision }}
+          path: ${{ github.workspace }}/SourceCache/swift-llbuild
+          show-progress: false
+      - uses: actions/checkout@v4
+        with:
+          repository: swiftlang/swift-tools-support-core
+          ref: ${{ inputs.swift_tools_support_core_revision }}
+          path: ${{ github.workspace }}/SourceCache/swift-tools-support-core
+          show-progress: false
+      - uses: actions/checkout@v4
+        with:
+          repository: apple/swift-argument-parser
+          ref: ${{ inputs.swift_argument_parser_revision }}
+          path: ${{ github.workspace }}/SourceCache/swift-argument-parser
+          show-progress: false
+      - uses: actions/checkout@v4
+        with:
+          repository: jpsim/Yams
+          ref: ${{ inputs.yams_revision }}
+          path: ${{ github.workspace }}/SourceCache/Yams
+          show-progress: false
+      - uses: actions/checkout@v4
+        with:
+          repository: swiftlang/swift-driver
+          ref: ${{ inputs.swift_driver_revision }}
+          path: ${{ github.workspace }}/SourceCache/swift-driver
+          show-progress: false
+
+      - uses: compnerd/gha-setup-vsdevenv@main
```

Comment:
Is it safe to unconditionally have this on Darwin?
IOW, do we want to add `if: inputs.build_os == 'Darwin'` for `gha-setup-vsdevenv`?

Reply:
This came up before, @compnerd said that this was a no-op on Mac and we should leave it as-is. See this [discussion](https://github.com/compnerd/swift-build/pull/806#discussion_r1722087115)